.github/workflows/coverage-report.yml

```diff
@@ -95,7 +95,7 @@ jobs:
           path: repo_root
 
       - name: Upload coverage to Codecov
-        uses: codecov/codecov-action@v3
+        uses: codecov/codecov-action@v4
```

Comment:
Should this maybe be `v4.1`, if `v4.0` has the bug?

Reply:
`codecov-action` maintainers seem to move `v4` tag as they release new minor versions. `v4` points to `v4.1.0` according to [this](https://github.com/codecov/codecov-action/tree/v4). I think it will be simpler to have it as v4 so we don't need to change when v4.2 is released.

Reply:
> Should this maybe be `v4.1`, if `v4.0` has the bug?

Good question!

> `codecov-action` maintainers seem to move `v4` tag as they release new minor versions. `v4` points to `v4.1.0` according to [this](https://github.com/codecov/codecov-action/tree/v4).

Yes, this is the way tags for GitHub actions typically work: people move them around so that you get the latest semver-compatible version when you depend on a given tag. It's a bit silly, but that's the way things work last time I checked.

The GitHub documentation still recommends this: [Using tags for release management](https://docs.github.com/en/actions/creating-actions/about-custom-actions#using-tags-for-release-management).